lib/options-manager.js

```diff
@@ -199,8 +207,8 @@ const findApplicableOverrides = (path, overrides) => {
 
 const mergeApplicableOverrides = (baseOptions, applicableOverrides) => {
 	applicableOverrides = applicableOverrides.map(normalizeOpts);
-	const overrides = [emptyOptions(), baseOptions].concat(applicableOverrides);
-	return deepAssign.apply(null, overrides);
+	const overrides = [emptyOptions(), baseOptions].concat(applicableOverrides, mergeFn);
+	return mergeWith.apply(null, overrides);
```

Comment:
Any reason why we aren't returning `mergeWith(emptyOptions(), baseOptions, applicableOverrides, mergeFn`?

Reply:
Oh, because `applicableOverrides` is an `Array`.